storage/s3/main.tf

```diff
@@ -1,3 +1,90 @@
+resource "aws_cloudwatch_log_group" "s3_cloudtrail_logs" {
+  count = var.s3_cloudwatch_logging_enabled ? 1 : 0
+  name  = "s3-cloudtrail-logs-${var.s3_bucket_name}"
+}
+
+data "aws_iam_policy_document" "cloudtrail-assume-role-policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+  }
+}
+
+
+
+resource "aws_s3_bucket" "s3_cloudtrail_logs" {
+
+  count  = var.s3_cloudwatch_logging_enabled ? 1 : 0
+  bucket = "${var.s3_bucket_name}-cloudtrail"
+  tags   = var.tags
+
+  policy = <<POLICY
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "AWSCloudTrailAclCheck",
+            "Effect": "Allow",
+            "Principal": {
+              "Service": "cloudtrail.amazonaws.com"
+            },
+            "Action": "s3:GetBucketAcl",
+            "Resource": "arn:aws:s3:::${var.s3_bucket_name}-cloudtrail"
+        },
+        {
+            "Sid": "AWSCloudTrailWrite",
+            "Effect": "Allow",
+            "Principal": {
+              "Service": "cloudtrail.amazonaws.com"
+            },
+            "Action": "s3:PutObject",
+            "Resource": "arn:aws:s3:::${var.s3_bucket_name}-cloudtrail/*",
+            "Condition": {
+                "StringEquals": {
+                    "s3:x-amz-acl": "bucket-owner-full-control"
+                }
+            }
+        }
+    ]
+}
+POLICY
+}
+
+
+resource "aws_iam_role" "cloudtrail_to_cloudwatch" {
+  count = var.s3_cloudwatch_logging_enabled ? 1 : 0
+  name  = "CloudWatchWriteForCloudTrail"
+
+  # Terraform's "jsonencode" function converts a
+  # Terraform expression result to valid JSON syntax.
+  assume_role_policy = data.aws_iam_policy_document.cloudtrail-assume-role-policy.json
+  inline_policy {
+    name = "cloudwatch_write_permissions"
+
+    policy = jsonencode({
+      Version = "2012-10-17"
+      Statement = [
+        {
+          Action = [
+            "cloudwatch:PutMetricData",
+            "logs:PutLogEvents",
+            "logs:DescribeLogStreams",
+            "logs:DescribeLogGroups",
+            "logs:CreateLogStream",
+            "logs:CreateLogGroup"
+          ]
+          Effect   = "Allow"
+          Resource = "*"
+        },
+      ]
+    })
+  }
+}
+
 resource "aws_s3_bucket" "kubeflow" {
```

Comment:
why we use `kubeflow` for the bucket resource name?

Reply:
it looks like this module appeared from sak-kubeflow and the changes proposed to us. We need to make this module more universal so that it can be used in various situations, and not just in the kubeflow installation

Reply:
should I update this PR , or we can open other issue ?

Reply:
Just update this, please. Could you write an example in Readme.md how to quickly add this module into the main project?

Reply:
yes, I will write a readme as well

Reply:
Check now